File: Java/src/main/java/QuestionBank/Solution201_300/Solution283.java
package QuestionBank.Solution201_300;

/**
 * @author windows
 */
public class Solution283 {

    public void moveZeroes(int[] nums) {
        int right = nums.length - 1;
        for (int i = 0; i < nums.length; i++) {
            if (nums[i] == 0){
                for (int j = i + 1; j < nums.length; j++) {
                    if (nums[j] != 0){
                        int temp = nums[i];
                        nums[i] = nums[j];
                        nums[j] = temp;
                        right = j;
                        break;
                    }
                }
            }
        }
    }


    public static void main(String[] args) {
        Solution283 solution283 = new Solution283();
    }
}

File: C++/src/Solution201_300/Solution283.cpp
//
// Created by windows on 2023/3/1.
//

#include "Solution283.h"
#include <functional>
#include <iostream>
#include <vector>
#include <unordered_set>
#include <unordered_map>
#include <string>
using namespace std;

void Solution283::moveZeroes(vector<int> &nums) {
    int i = 0;
    for (int j = 0; j < nums.size(); ++j) {
        if (nums[j] != 0) {
            int temp = nums[j];
            nums[j] = nums[i];
            nums[i++] = temp;
        }
    }
}
